src/maps/maps.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, Logger, HttpStatus } from '@nestjs/common';
+import { PrismaService } from '../prisma.service';
+import { ResponseSmokingArea } from './dto/read-map.dto';
+import { HttpService } from '@nestjs/axios';
+import CustomException from 'src/exceptions/custom.exception';
+import { firstValueFrom } from 'rxjs';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class MapsService {
+  constructor(
+    private prisma: PrismaService,
+    private readonly http: HttpService,
+    private readonly config: ConfigService,
+  ) {}
+
+  private readonly logger = new Logger(MapsService.name);
+
+  async getMapById(
+    mapId: number,
+    lat: number,
+    lang: number,
+  ): Promise<ResponseSmokingArea> {
+    try {
+      const map = await this.prisma.map.findUnique({
+        where: {
+          id: mapId,
+        },
+      });
+
+      if (!map) {
+        throw new CustomException(HttpStatus.NOT_FOUND, 'Not Found - Map');
+      }
+
+      const defaultImage = this.config.get('DEFAULT_IMAGE');
+
+      const { latitude: destinationLatitude, longitude: destinationLongitude } =
+        map;
+
+      const osrmRouteUrl = `http://router.project-osrm.org/route/v1/driving/${lang},${lat};${destinationLongitude},${destinationLatitude}`;
+      const osrmResponse = await firstValueFrom(this.http.get(osrmRouteUrl));
+      const distance = `${osrmResponse.data.routes[0].distance}m` ?? '-m';
```

Comment:
<p style="box-sizing: border-box; margin: 0px 0px 20px; padding: 0px; border: 0px; font-size: 15px; vertical-align: baseline; display: block; overflow-wrap: break-word; color: rgba(0, 0, 0, 0.75); font-family: sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">In case of error the following<span> </span><code style="box-sizing: border-box; margin: 0px; padding: 3px; border: 0px; font-size: 12px; vertical-align: baseline; font-family: Menlo, &quot;Bitstream Vera Sans Mono&quot;, Monaco, Consolas, monospace; white-space: pre-wrap; background: rgba(0, 0, 0, 0.024); line-height: 20px; border-radius: 3px;">code</code>s are supported in addition to the general ones:</p>

Type | Description
-- | --
NoRoute | No route found.

<p style="box-sizing: border-box; margin: 0px; padding: 0px; border: 0px; font-size: 15px; vertical-align: baseline; display: block; overflow-wrap: break-word; color: rgba(0, 0, 0, 0.75); font-family: sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">All other properties might be undefined.</p>

---

OSRM의 Route service가 원래는 두 좌표 사이의 가장 빠른 경로를 찾는 api여서 오류의 경우 '경로 없음'이라고만 반환값이 온다고 합니다. 우선은 이런 경우 '-m'로 처리했습니다. 

추후 직선거리라도 계산해서 주는게 좋을지 논의해보면 좋을 것 같아요

Reply:
일단은 이렇게 개발하고 기획단이나 아요랑 같이 논의해봐야겠네요! 찾아보느라 고생하셨습니다.